src/libstore/meson.build

```diff
@@ -0,0 +1,202 @@
+# libstore build file
+#============================================================================
+
+libstore_dir =  meson.current_source_dir()
+libstore_inc = [include_directories('.')]
+
+# dependencies
+#============================================================================
+
+libstore_dep_list = [
+    aws_sdk_cpp_dep,
+    libbz2_dep,
+    libcurl_dep,
+    libdl_dep,
+    libseccomp_dep,
+    libsodium_dep,
+    nlohmann_dep,
+    pthread_dep,
+    sqlite3_dep,
+
+    libutil_dep]
+
+
+# src files
+#============================================================================
+
+libstore_src = files(
+    join_paths(libstore_dir, 'binary-cache-store.cc'),
+    join_paths(libstore_dir, 'content-address.cc'),
+    join_paths(libstore_dir, 'crypto.cc'),
+    join_paths(libstore_dir, 'daemon.cc'),
+    join_paths(libstore_dir, 'derivations.cc'),
+    join_paths(libstore_dir, 'dummy-store.cc'),
+    join_paths(libstore_dir, 'export-import.cc'),
+    join_paths(libstore_dir, 'filetransfer.cc'),
+    join_paths(libstore_dir, 'gc.cc'),
+    join_paths(libstore_dir, 'globals.cc'),
+    join_paths(libstore_dir, 'http-binary-cache-store.cc'),
+    join_paths(libstore_dir, 'legacy-ssh-store.cc'),
+    join_paths(libstore_dir, 'local-binary-cache-store.cc'),
+    join_paths(libstore_dir, 'local-fs-store.cc'),
+    join_paths(libstore_dir, 'local-store.cc'),
+    join_paths(libstore_dir, 'lock.cc'),
+    join_paths(libstore_dir, 'machines.cc'),
+    join_paths(libstore_dir, 'misc.cc'),
+    join_paths(libstore_dir, 'names.cc'),
+    join_paths(libstore_dir, 'nar-accessor.cc'),
+    join_paths(libstore_dir, 'nar-info.cc'),
+    join_paths(libstore_dir, 'nar-info-disk-cache.cc'),
+    join_paths(libstore_dir, 'optimise-store.cc'),
+    join_paths(libstore_dir, 'parsed-derivations.cc'),
+    join_paths(libstore_dir, 'path.cc'),
+    join_paths(libstore_dir, 'pathlocks.cc'),
+    join_paths(libstore_dir, 'profiles.cc'),
+    join_paths(libstore_dir, 'realisation.cc'),
+    join_paths(libstore_dir, 'references.cc'),
+    join_paths(libstore_dir, 'remote-fs-accessor.cc'),
+    join_paths(libstore_dir, 'remote-store.cc'),
+    join_paths(libstore_dir, 's3-binary-cache-store.cc'),
```

Comment:
This fails to compile without the S3 SDK as a dependency, for which the option (`with_s3`) is set to `false` by default here...
Feels like this file should be added to the sources iif the option is `true`.

Reply:
this was a problem, see https://github.com/NixOS/nix/pull/4734#event-4824215224

Reply:
it needs to be defined if enabled, undefined if not. couldnt be implemented like that due to conflicts, so the code would only compile if it was disabled.

Reply:
As it is now, I can't get `nixstore` to compile...
I tried with `with_s3=false`, without better success :confused:

Reply:
iirc it's not with_s3=false its -Dwith_s3=false, which may be a help. also as i said, the code needs to be changed in the aws dep file to match the other dependency styles. instead of defining enable_s3 0 at the bottom of that dep, you need to remove the entry into the config file entirely.

Reply:
comment out these lines :) 

https://github.com/p01arst0rm/nix/blob/4c43170c7ab289ecb567f4f987e0f9e37ec859a9/dependencies/aws-sdk-cpp/meson.build#L45-L47